src/Extensions/FigureGenerator.tsx

```diff
@@ -233,3 +243,23 @@ export const setupFigure = (gameBoard: number[][], figure: Figure) => {
 
   return updatedGameBoard;
 };
+
+const findFilledRow = (gameBoard: number[][]) => {
+  return gameBoard.findIndex((item, index) => {
+    return item.every((cell) => cell === 1);
+  });
+};
+
+export const deleteFilledRows = (gameBoard: number[][]) => {
+  let updatedGameBoard = copyGameBoard(gameBoard);
+
+  for (let row = 0; row < PLAYGROUND_HEIGHT; row++) {
+    let index = findFilledRow(updatedGameBoard);
+    if (index !== -1) {
+      updatedGameBoard.splice(index, 1);
+      updatedGameBoard.unshift([0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
+    }
+  }
```

Comment:
This doesn't look good. The loop should be `do { ... } while (index !== 1)` but on each iteration it calculate all the filled rows again. Better solution is to 
```suggestion
  for (let row = 0; row < PLAYGROUND_HEIGHT; row++) {
    if (isRowFilled(updatedGameBoard[row])) {
      updatedGameBoard.splice(row, 1);
      updatedGameBoard.unshift([0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
      row++;
    }
  }
```